src/client/thingtypemanager.cpp

```diff
@@ -237,6 +321,28 @@ ThingTypeList ThingTypeManager::findThingTypeByAttr(const ThingAttr attr, const
     return ret;
 }
 
+const RaceType& ThingTypeManager::getRaceData(uint32_t raceId)
+{
+    for (const auto& raceData : m_monsterRaces) {
+        if (raceData.raceId == raceId) {
+            return raceData;
+        }
+    }
+    auto emptyVal = new RaceType();
+    return *emptyVal;
+}
+
+const std::vector<RaceType>& ThingTypeManager::getRacesByName(const std::string& searchString)
+{
+    static std::vector<RaceType> result;
```

Comment:
Is it expected for getRacesByName to return the same result across multiple calls? Since result is static, it retains data from previous calls, which could lead to bugs if the function is called multiple times with different input. If that's not the intended behavior, consider using a local variable instead of a static one to ensure the results are independent for each call.

Reply:
I was trying various ways of pushing this to Lua. Overlooked this one when adapting the code. Thanks!